libbeat/docs/monitoring/monitoring-beats.asciidoc

```diff
@@ -14,18 +14,18 @@ ifdef::apm-server[]
 endif::[]
 
 To monitor {beatname_uc}, make sure monitoring is enabled on your {es} cluster,
-then configure the method used to collect {beatname_uc} metrics. You
-ifndef::serverless[]
-can use one of following methods:
-endif::[]
-ifdef::serverless[]
-can use the following method:
-endif::[]
+then configure the method used to collect {beatname_uc} metrics. You can use one
+of following methods:
```

Comment:
I think it's fine in the case of Beats to be relatively unopinionated about which method one should use for monitoring collection, but in reading this, I'm left with the feeling that a user may get frustrated by not knowing which is best for their use case. Perhaps we should expand this section a bit to either:

1) Recommend one option over the other.

or

2) Give more guidance about why one should choose one over the other.

Thoughts?

Reply:
I agree, but I'm not sure we've decided what to recommend yet because it depends on how we plan to implement monitoring in agent. The upgrade to agent might be simpler in the future if folks use internal collection now.

Reply:
> 2. Give more guidance about why one should choose one over the other.

In the interest of moving this PR forward, how about we go with this option for now? If/when we decide strongly on a recommendation, we can do a follow up PR then.

IMO the benefit of monitoring Beats (or any other Elastic product) with Metricbeat is that you get an external agent doing the monitoring, which could be more robust, in that if the monitored instance dies, the agent doing the monitoring is still up.

However, the benefit of using internal collection is one less piece of software to install and maintain. This is really where having Agent will be nice, as it will encapsulate the Beat(s) being monitored as well as the Metricbeat monitoring these Beat(s).